src/ocean/io/serialize/StringStructSerializer.d

```diff
@@ -345,18 +345,19 @@ public class StringStructSerializer ( Char )
 
         Params:
             output = string to serialize struct data to
-            array = array to append
             name = name of array item
+            array = array to append
+            typename = name of the baset type (defaults to T.stringof)
 
     ***************************************************************************/
 
     public void serializeArray ( T ) ( ref Char[] output, cstring name,
-        T[] array )
+        T[] array, cstring typename = T.stringof )
```

Comment:
I'd also suggest small refactoring to make `serializeStructArray` sub-routine of `serializeArray` so that this awkward extra argument is not needed.

Reply:
I am not sure I understand what you are suggesting.
so you suggest putting typedef-handling logic into serializeStructArray ?

Reply:
Both array and struct array start with generating the very same string - `Hash[] x (length 3):` (the only one which needs "real" name). Branching for structs only happens after.

So it should be possible to just call `serializeArray` for both and do `static if (T == struct) serializeStructArrayContent()` inside it, where `serializeStructArrayContent` is new imaginary method to replace `serializeStructArray`.